pkg/volume/csimigration/patch_adc_plugin_manager.go

```diff
@@ -0,0 +1,42 @@
+package csimigration
+
+import (
+	"k8s.io/component-base/featuregate"
+	csilibplugins "k8s.io/csi-translation-lib/plugins"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+// NewADCPluginManager returns a new PluginManager instance for the Attach Detach controller which uses different
+// featuregates in openshift to control enablement/disablement which *DO NOT MATCH* the featuregates for the rest of the
+// cluster.
+func NewADCPluginManager(m PluginNameMapper, featureGate featuregate.FeatureGate) PluginManager {
```

Comment:
I only saw 6 lines that use the `CSIMigrationvSphere`, could we simply add ` && features.OpenshiftStartCSMigrationVSphere()` to those 6 lines?

Reply:
This code comes from [our carry-patch we had in 4.8](https://github.com/openshift/kubernetes/pull/601) (or 4.9?), it's only cleaned to handle only vSphere. It makes sure that A/D controller sees the CSI migration as always enabled, so we don't need to update KCM first and then rest of the cluster. See the linked PR for details.

Reply:
I think it could be better to separate it to a standalone PR.